BUILD.md

```diff
@@ -0,0 +1,472 @@
+:balloon: Welcome to the Conduit development guide! :wave:
+
+This document will help you build Conduit from source. Depending on the
+components you are interested in building, these instructions assume you have
+the following tools already setup:
+- [Go](https://golang.org)
+- [Rust](https://www.rust-lang.org)
+- [Minikube](https://github.com/kubernetes/minikube)
+- [Docker](https://www.docker.com/)
+- [Yarn](https://yarnpkg.com)
+
+# Repo layout
+
+Conduit is primarily writtern in Go and Rust (and JavaScript!). At its core
+is a high-performance data plane written in Rust. The control plane components
+are writtern in Go. The dashboard UI is a React application.
+
+## Control Plane (Go)
+
+- [`cli`](cli): Command-line `conduit` utility, view and drive the control
+  plane.
+- [`controller`](controller)
+  - [`destination`](controller/destination): Serves service discovery
+    information to the `proxy`.
+  - [`proxy-api`](controller/api/proxy): Accepts requests from `proxy`
+    instances and forwards those requests to the appropriate controller
+    service.
+  - [`public-api`](controller/api/public): Accepts requests from API
+    clients such as `cli` and `web`, provides access to and control of the
+    conduit service mesh.
+  - [`tap`](controller/tap): Provides a live pipeline of requests.
+  - [`telemetry`](controller/telemetry): Collects and aggregates
+    metrics from `proxy` componenets.
+- [`proxy-init`](proxy-init): Adds a Kubernetes pod to join the Conduit
+  Service Mesh.
+- [`web`](web): Provides a UI dashboard to view and drive the control plane.
+
+## Data Plane (Rust)
+
+  - [`proxy`](proxy): High-performance data plane, injected as a sidecar with
+    every service.
+  - [`tower-grpc`](tower-grpc): A client and server gRPC implementation based on
```

Comment:
Note that we are hoping to stop vendoring these library dependencies and depend on them from their Git repos --- see #119. Hopefully this will happen soon.